src/metatrain/__main__.py

```diff
@@ -83,7 +83,15 @@ def main():
         # define and create `checkpoint_dir` based on current directory, date and time
         checkpoint_dir = _datetime_output_path(now=datetime.now())
         if is_main_process():
-            os.makedirs(checkpoint_dir)
+            if not os.path.exists(checkpoint_dir):
+                os.makedirs(checkpoint_dir)
+            else:
+                # directory already exists in a different run, add a suffix
+                # (.1, .2, ...) to the directory name
+                i = 1
+                while os.path.exists(f"{checkpoint_dir}.{i}"):
+                    i += 1
+                checkpoint_dir = Path(f"{checkpoint_dir}.{i}")
```

Comment:
I think this still has the same issue: a separate directory with the same name could still be created between the check and the directory creation (https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use). 

How about something like

```suggestion
            try:
                os.makedirs(checkpoint_dir)
            except FileExistsError:
                # directory already exists from a different run, add a suffix
                # (.1, .2, ...) to the directory name
                initial_checkpoint_dir = checkpoint_dir
                i = 1
                while True:
                    try:
                        checkpoint_dir = f"{initial_checkpoint_dir}.{i}"
                        os.makedirs(checkpoint_dir)
                        break
                    except FileExistsError:
                        i += 1
```

Reply:
You're right, good idea